app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessagePollingService.java

```diff
@@ -123,9 +126,14 @@ private void handleActionMessagePolling() {
     private void processNewMessages(Collection<Message> messages) {
         Log.v(LOG_TAG, "MessagePolling service processing " + messages.size() + " new messages");
 
-        for (Message m :
+        for (Message msg :
                 messages) {
-            MessagePubSub.getInstance().publish(m);
+            if (msg.getSenderId().equals(NetworkUtil.getMac())) {
+                continue;
+            }
+            Intent intent = new Intent(msg.getClass().getName());
```

Comment:
these 3 lines can be a static method in MessageBroadcastReceiver:
static void broadcastMessage(Message msg)